src/algorithms/klminnaturalgraddescent.jl

```diff
@@ -0,0 +1,175 @@
+- The target distribution has unconstrained support (\$\$\\mathbb{R}^d\$\$).
+- The target `LogDensityProblems.logdensity(prob, x)` has at least first-order differentiation capability.
+"""
+@kwdef struct KLMinNaturalGradDescent{Sub<:Union{Nothing,<:AbstractSubsampling}} <:
+              AbstractVariationalAlgorithm
+    stepsize::Float64
+    n_samples::Int = 1
+    ensure_posdef::Bool = true
+    subsampling::Sub = nothing
+end
+
+struct KLMinNaturalGradDescentState{Q,P,S,Prec,GradBuf,HessBuf}
+    q::Q
+    prob::P
+    prec::Prec
+    iteration::Int
+    sub_st::S
+    grad_buf::GradBuf
+    hess_buf::HessBuf
+end
+
+function init(
+    rng::Random.AbstractRNG,
+    alg::KLMinNaturalGradDescent,
+    q_init::MvLocationScale{<:LowerTriangular,<:Normal,L},
+    prob,
+) where {L}
+    sub = alg.subsampling
+    n_dims = LogDensityProblems.dimension(prob)
+    capability = LogDensityProblems.capabilities(typeof(prob))
+    if capability < LogDensityProblems.LogDensityOrder{1}()
+        throw(
+            ArgumentError(
+                "`KLMinNaturalGradDescent` requires at least first-order differentiation capability. The capability of the supplied `LogDensityProblem` is $(capability).",
+            ),
+        )
+    end
+    sub_st = isnothing(sub) ? nothing : init(rng, sub)
+    grad_buf = Vector{eltype(q_init.location)}(undef, n_dims)
+    hess_buf = Matrix{eltype(q_init.location)}(undef, n_dims, n_dims)
+    return KLMinNaturalGradDescentState(
+        q_init, prob, inv(cov(q_init)), 0, sub_st, grad_buf, hess_buf
+    )
+end
+
+output(::KLMinNaturalGradDescent, state) = state.q
+
+function step(
+    rng::Random.AbstractRNG,
+    alg::KLMinNaturalGradDescent,
+    state,
+    callback,
+    objargs...;
+    kwargs...,
+)
+    (; ensure_posdef, n_samples, stepsize, subsampling) = alg
+    (; q, prob, prec, iteration, sub_st, grad_buf, hess_buf) = state
+
+    m = mean(q)
+    S = prec
+    η = convert(eltype(m), stepsize)
+    iteration += 1
+
+    # Maybe apply subsampling
+    prob_sub, sub_st′, sub_inf = if isnothing(subsampling)
+        prob, sub_st, NamedTuple()
+    else
+        batch, sub_st′, sub_inf = step(rng, subsampling, sub_st)
+        prob_sub = subsample(prob, batch)
+        prob_sub, sub_st′, sub_inf
+    end
+
+    logπ_avg, grad_buf, hess_buf = gaussian_expectation_gradient_and_hessian!(
+        rng, q, n_samples, grad_buf, hess_buf, prob_sub
+    )
+
+    S′ = Hermitian(((1 - η) * S + η * Symmetric(-hess_buf)))
+    if ensure_posdef
+        G_hat = S - Symmetric(-hess_buf)
+        S′ += η^2 / 2 * Symmetric(G_hat * (S′ \ G_hat))
+    end
+    m′ = m - η * (S′ \ (-grad_buf))
+
+    q′ = MvLocationScale(m′, inv(cholesky(S′).L), q.dist)
+
+    state = KLMinNaturalGradDescentState(
+        q′, prob, S′, iteration, sub_st′, grad_buf, hess_buf
+    )
+    elbo = logπ_avg + entropy(q′)
+    info = merge((elbo=elbo,), sub_inf)
+
+    if !isnothing(callback)
+        info′ = callback(; rng, iteration, q, info)
```

Comment:
```suggestion
        info′ = callback(; rng, iteration, q′, info)
```

Reply:
Oops. Let me fix this in the other algorithms as well